src/amqp.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#ifndef  _GNU_SOURCE
+#define  _GNU_SOURCE
+#endif
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <ell/ell.h>
+#include <hal/linux_log.h>
+#include <amqp.h>
+#include <amqp_tcp_socket.h>
+
+#include "settings.h"
+#include "amqp.h"
+
+#define AMQP_CONNECTION_TIMEOUT_US 10000
+
+static amqp_connection_state_t conn;
+static struct amqp_connection_info cinfo;
```

Comment:
Why defining it out of `int amqp_start(struct settings *settings)`?

Reply:
Once that cinfo is assigned there is no reason to change it. And we can use this struct to other function like handle disconnections and retry connection

Reply:
I agree with you that it might be useful for connection retrial. But, if it was being used to do so, this `retry_connection()` function would be called by `amqp_start()`, wouldn't it?

Reply:
I also agree that the info is not to be changed after it is assigned. But there is also no reason for it to remain stored.

By defining it as a static variable, it will not be discarded, occupying memory for nothing.

Reply:
Not necessarily it will be called by `amqp_start`. With libell we can set a callback when a socket is disconnected.
The `msg.c` can call a `amqp_connect` in a callback when receives a disconnection. And if we store neither `settings->rabbitmq_url` nor `cinfo` we will be unable to reconnect again. Does that make sense?

Reply:
If you think that is more suitable just make this change when this connection handler is implemented I can put it as a local variable too. But I think that it will happen in the future patches.

Reply:
I see. Makes sense

Reply:
But I only see a reason for making it static when this said function is implemented.

Reply:
It's a simple change that can be done when necessary